pkg/agent/agent_test.go

```diff
@@ -1966,3 +2037,66 @@ func updateActionToNode(t *testing.T, action k8stesting.Action) *corev1.Node {
 
 	return node
 }
+
+// lifted from https://github.com/kubernetes/kubectl/blob/master/pkg/drain/drain_test.go.
+func addEvictionSupport(t *testing.T, clientset *fake.Clientset, version string) {
+	t.Helper()
+
+	podsEviction := metav1.APIResource{
+		Name:    "pods/eviction",
+		Kind:    "Eviction",
+		Group:   "policy",
+		Version: version,
+	}
+	coreResources := &metav1.APIResourceList{
+		GroupVersion: "v1",
+		APIResources: []metav1.APIResource{podsEviction},
+	}
+
+	policyResources := &metav1.APIResourceList{
+		GroupVersion: "policy/v1",
+	}
+	clientset.Resources = append(clientset.Resources, coreResources, policyResources)
+
+	// Delete pods when evict is called
+	clientset.PrependReactor("create", "pods", func(action k8stesting.Action) (bool, runtime.Object, error) {
+		if action.GetSubresource() != "eviction" {
+			return false, nil, nil
+		}
+
+		namespace := ""
+		name := ""
+		switch version {
+		case "v1":
+			var eviction *policyv1.Eviction
+			if a, ok := action.(k8stesting.CreateAction); ok {
+				if p, ok := a.GetObject().(*policyv1.Eviction); ok {
+					eviction = p
+				}
+			}
+			namespace = eviction.Namespace
+			name = eviction.Name
+		case "v1beta1":
+			var eviction *policyv1beta1.Eviction
+			if a, ok := action.(k8stesting.CreateAction); ok {
+				if p, ok := a.GetObject().(*policyv1beta1.Eviction); ok {
+					eviction = p
+				}
+			}
+			namespace = eviction.Namespace
+			name = eviction.Name
+		default:
+			t.Errorf("unknown version %s", version)
+		}
+		// Avoid the lock
```

Comment:
If we keep this comment, it should end with dot.
```suggestion
		// Avoid the lock.
```